src/screens/registerScreen.js

```diff
@@ -7,12 +7,20 @@ const registerScreen = (
   Component,
   store,
 ) => {
-  const ConnectedComponent = props =>
-    <Provider store={store}>
-      <Component
-        {...props}
-      />
-    </Provider>;
+  class ConnectedComponent extends PureComponent {
```

Comment:
Why not keep it as a function component?

```suggestion
  const ConnectedComponent = props =>
    <Provider store={store}>
      <Component {...props} />
    </Provider>;
  ConnectedComponent.options = Component.options;
```